pennylane/qchem/observable_hf.py

```diff
@@ -52,6 +52,7 @@ def fermionic_observable(constant, one=None, two=None, cutoff=1.0e-12):
     + 0.5 * a⁺(3) a(3)
     """
     coeffs = qml.math.array([])
+    constant = qml.math.array([constant])
```

Comment:
Why is this needed?

Reply:
Context:
I think this is due to a change I made in `_nuclear_energy`, which I've changed cause of jax reasons (Basically jax wasn't happy `_nuclear_energy` was returning an array instead of a scalar when I tried `jax.grad()` on `nuclear_energy`). 

Therefore, constant turned from an array to a scalar. Buuuuut, if you don't coerce constant into an array for `observable_hf` though you get a dimension error when you concat. But I think there is a better/smarter solution for this particular thing but for now this works so I'm leaving it like this while I fix the other bugs.